### ISSUE TITLE
Assorted issues with the new aircraft crashsites

### ISSUE THREAD

**Issue by Regularitee** [CleverRaven/Cataclysm-DDA#25143]
- [ ] Found a human stomach, flesh, and bone, but no corpse strapped to a seat. 
- [ ] Clothing quality of said missing person was all perfect condition, all undamaged and clean
- [ ] Metal wreckage spawns underneath the wreck. This is a pain since you cannot visually see it, which makes climbing inside a vehicle a blind trial-and-error maze of "Do you really want to step onto metal wreckage?". It also mean zeds you kill inside phase through the vehicle floor and get magically buried under said wreckage, leaving inaccessible corpses when killed.
- [ ] I just fixed vehicle weapon weights (#25037) and you go and add a bunch of weapons/ammo that are way off their real values! REEEEE. Fix plz. (Most are just a quick wikipedia check away)
- [ ] Aircraft spawn with a full crew. Literally no other vehicles in the game do this, which I find a bit odd...
- [ ] When examining downed aircraft, you can see the full shape (including missing sections?) in the vehicle info window. This never happens for normal city wrecks, so I assume it's a bug?
- [ ] Some aircraft appear to be converted to nuclear power, yet use car batteries for storage?

**Comment by CoroNaut** [comment]
Are we assuming that every aircraft crash will kill all occupants? It would be cool to get an npc like 5% of the time.

**Comment by Xenomorph-III** [comment]
> Are we assuming that every aircraft crash will kill all occupants? It would be cool to get an npc like 5% of the time.

With various wounds and a pain modifier,, like the player? Yes, I think it would be great.

**Comment by FulcrumA** [comment]
>Found a human stomach, flesh, and bone, but no corpse strapped to a seat.

Limitation of the current vehicles. Bones etc are a workaround. To allow corpses, there'd have to be certain changes to how vehicle tiles handle items. In the initial version (the one before merge) the vehicle I've made had corpses but since they were above the volume limit of the seat, they didn't spawn with their loot.

>Clothing quality of said missing person was all perfect condition, all undamaged and clean

Another thing that bothered me, too. When I exported the design, the condition of the item apparently wasn't tracked with the corpse. I've looked into it since then a bit and perhaps I'll be able to improve things a bit here.

>Metal wreckage spawns underneath the wreck. This is a pain since you cannot visually see it, which makes climbing inside a vehicle a blind trial-and-error maze of "Do you really want to step onto metal wreckage?". It also mean zeds you kill inside phase through the vehicle floor and get magically buried under said wreckage, leaving inaccessible corpses when killed.

The zombies buried under wreckage is, I am afraid, a thing from even before I even had an idea to start working on the crash sites. Same with the tiles of metal wreckage itself - though personally, I find it to be an unplanned feature, what's with moving through a wreckage full of jagged, dangerous bits sticking out. Again, some of it probably would require some code changes - not to say that I wouldn't like such a change, just - not up to me as the vehicle person.

>I just fixed vehicle weapon weights (#25037) and you go and add a bunch of weapons/ammo that are way off their real values! REEEEE. Fix plz. (Most are just a quick wikipedia check away)

Nope. I assure you, I've added no parts, merely used what's already in the base game. If I'd be really ambitious (and it may happen) and I would be adding new parts I'd primarily concentrate on stuff like rotors/rotor blades, inbuilt black box part rather than current workaround and similar things.

>Aircraft spawn with a full crew. Literally no other vehicles in the game do this, which I find a bit odd...

The crew spawn should be absolutely randomized. During initial tests even empty helis could be found. Pinging @mrkybe as he handled the coding side of our little project.

>When examining downed aircraft, you can see the full shape (including missing sections?) in the vehicle info window. This never happens for normal city wrecks, so I assume it's a bug?

That is weird. Pre-exported vehicles appear absolutely normal in each variation.

>Some aircraft appear to be converted to nuclear power, yet use car batteries for storage?

Welp, that's my fault. Probably remnant from some tests I was performing and forgot about. Found another such tidbit that doesn't fit there, will remove it when doing loot changes, thanks for bringing it up.

**Comment by FulcrumA** [comment]
@mlangsdorf 
To the extend as I've addressed it here (which means - not fully):
>no corpse strapped to a seat

Currently impossible without changing how seats work/how much they can contain which would affect vehicles in general and thus I am unsure if it should be changed. That's the reason for my human organs workaround.

>Clothing quality of said missing person was all perfect condition, all undamaged and clean

Clothes are now damaged, however unless I've missed something, "flags": [ "FILTHY" ] cannot be applied to regular item spawn. Otherwise they're the same clothes a zombie would have now, with slight adjustments.

>Metal wreckage spawns underneath the wreck.

Metal wreckage under vehicle is also a thing not really part of heli crashes themselves as much as a mechanics' behavior in general and not resolved. Personally I like it even though it's bothersome - makes it harder to navigate what is a wreckage. May need further discussion.

>you go and add a bunch of weapons/ammo that are way off their real values! REEEEE. Fix plz.

I didn't add any item that I didn't have in the unmodded vanilla so if it requires balancing, it's part of whatever vanilla gun rebalancing endeavor there was elsewhere.

>Aircraft spawn with a full crew. Literally no other vehicles in the game do this, which I find a bit odd...

Aircrafts spawn with randomized number of (former) crew, including 0. @mrkybe probably would have to be the one to balance it further in the direction fo that 0 if it still spawns too many.

>When examining downed aircraft, you can see the full shape (including missing sections?)

Couldn't reproduce. Spawns alright for me.

>Some aircraft appear to be converted to nuclear power, yet use car batteries for storage?

Some redundant parts like a car battery leftover from some testing are now removed.
